src/modules/auth/auth.service.ts

```diff
@@ -82,13 +89,54 @@ export class AuthService {
     const accessToken = await this.generateAuthToken(payload, 'access');
     const refreshToken = await this.generateAuthToken(payload, 'refresh');
 
+    const refreshTokenEntity = this.createRefreshTokenEntity(refreshToken);
+    await this.refreshTokenRepository.insert(refreshTokenEntity);
+
     return {
       message: 'User logged in successfully',
       accessToken,
       refreshToken,
     };
   }
 
+  async issueNewTokens(refreshToken: string) {
+    const decodedRefreshToken =
+      this.jwtService.decode<AuthTokenPayload>(refreshToken);
+
+    const existingRefreshTokenEntity =
+      await this.refreshTokenRepository.findOne({
+        where: { userId: decodedRefreshToken.sub, token: refreshToken },
+        relations: {
+          user: true,
+        },
+      });
+
+    if (!existingRefreshTokenEntity) {
```

Comment:
Or add condition for **is_deleted** here

Reply:
> Or add condition for **is_deleted** here

Done